### PR TITLE
Hide thread tracer

### PR DESCRIPTION
**Pull request by vext01**
To aid the reviewer, I've put the main bulk of the work into the first commit. The commit message should explain the intent of the change.

The second commit is just adjustments to make tests pass again (and isn't all that interesting).

We can squash them together later.

We don't have (and can't easily make) tests for `ykrt` yet, so we will just have to eyeball those changes.

### CODE REVIEW

File: yktrace/src/lib.rs [hunk 1]
```diff
@@ -12,6 +13,15 @@ mod hwt;
 pub use errors::InvalidTraceError;
 pub use hwt::mapper::BlockMap;
 
+thread_local! {
+    // When `Some` contains the `ThreadTracer` for the current thread. When `None`, the current
+    // thread is not being traced.
+    //
+    // We hide the `ThreadTracer` in a thread local (rather than returning it to the consumer of
+    // yk). This ensures that the `ThreadTracer` itself cannot appear in traces.
+    pub static THREAD_TRACER: RefCell<Option<ThreadTracer>> = RefCell::new(None);
```
Comment (bjorn3) [inline yktrace/src/lib.rs:22]: There is a nightly feature that allows writing

```suggestion
    pub static THREAD_TRACER: RefCell<Option<ThreadTracer>> = const { RefCell::new(None) };
```

This makes TLS accesses cheaper as it doesn't have to check if it needs to lazily initialize it every time, instead evaluating the initializer at compile time and storing it's result in the `.tdata` section which libc copies to form the TLS storage for a thread.

**Comment by ltratt** [comment]
A couple of doc comments, but otherwise this looks OK!

File: yktrace/src/lib.rs (hunk 1 above)
Comment (vext01) [reply yktrace/src/lib.rs]: Thanks! I've implemented that in eac6196.

**Comment by vext01** [comment]
@ltratt your comments are addressed in 4800062.

**Comment by ltratt** [comment]
Please squash.

**Comment by vext01** [comment]
splat.

**Comment by ptersilie** [comment]
Looks good.

**Comment by vext01** [comment]
Ready to go @ltratt ?

**Pull request merged**